recipe/meta.yaml

```diff
@@ -219,7 +207,7 @@ outputs:
         - python
       run_constrained:
         - arrow-cpp-proc * {{ build_ext }}
-        - cudatoolkit >=9.2  # [cuda_compiler_version != "None"]
+        - cudatoolkit >={{ cuda_compiler_version_min }}  # [cuda_compiler_version != "None"]
```

Comment:
This was intentional since Arrow only uses the CUDA driver library

Reply:
Yes, that's why it's set to the minimum? I.e. aarch it's 11, and `cuda_compiler_version_min` takes care of that on a cf level?

Or can we be sure that something compiled with CUDA 10.2 is still compatible with CUDA 9?

Reply:
It doesn't use `nvcc` or any of the libraries in the `cudatoolkit` package. It only uses the CUDA driver library, `libcuda`, which is installed and versioned separately.

I believe this was added because `cudatoolkit` has a `__cuda` virtual package dependency to ensure that the driver version sufficiently supports the toolkit.

I'm not sure if there was a reason that we didn't use `__cuda` directly here.

Reply:
Could be because that virtual package wasn't around yet at the time when arrow added CUDA support (not sure exactly about the timing)... So let's just use `__cuda`?

Reply:
I'm not sure. It may have been weird solver behavior at the time or something else. Regardless, could we defer changing this to a follow up PR and just leave this as `cudatoolkit >=9.2` for now?

Reply:
I'll do it for 6.0.x and 7.0.x, but the same change is present on 8.0.x, 9.0.x and main. Could you open an issue to discuss resp. keep track of this?

Reply:
All of these changes were in #875 for more than a month. I just backported the relevant pieces to the maintenance branches afterwards.

In the case of these PRs, I _also_ gave several days to respond to https://github.com/conda-forge/arrow-cpp-feedstock/issues/918

Reply:
Understood. Though there is a quite a lot happening in that PR, which is likely how Keith missed this.

In any event am trying to make a minor procedural suggestion to limit surprise after the fact, which seems to be what is happening here.

Reply:
Which is why I put in a lot of effort to make it reviewable per-commit, and pinged several times for people to do so. So I don't agree that this is surprising, but well, we're all very busy and thing slip through the cracks.

In any case, I don't see any harm done here - aside from the fact that CUDA<10.2 is positively ancient, we can just loosen the pins again and voilà, packages are installable again even for those on CUDA 9.2.

Reply:
Understand you put a lot of effort into that PR and appreciate it

Only suggesting that when we have an unrelated change in PRs in the future, it would help to comment on them to identify that change and raise it for discussion. Could you please do this in the future?

Am not trying to discuss this specific change. Agree with you that is better handled in an issue

Reply:
@h-vetinari I truly appreciate all of the work you've been doing in helping to move this and other feedstocks forward. I was sick for ~3 weeks recently and truly out of commission and then was on vacation celebrating my 1st anniversary with my wife, so I've been MIA for quite a while and am still playing catch up.

I'm also perfectly happy to move forward as is here. With CUDA 12.0 having been released recently being back 3 major versions feels more than generous enough.

Reply:
> I was sick for ~3 weeks recently and truly out of commission and then was on vacation celebrating my 1st anniversary with my wife, so I've been MIA for quite a while and am still playing catch up.

Sorry to hear - hope you're feeling better now, and congrats on the anniversary 🥳

> With CUDA 12.0 having been released recently being back 3 major versions feels more than generous enough.

Not sure how long it'll be until we support 12 in conda-forge, but in any case, this should now be compatible with 9.2 again (so 9, 10, 11). Not sure how many people are still on drivers more than 3 years old; I mainly wanted to avoid broken metadata (because we're using 10.2 as minimum now, and loosening restrictions after the fact is always easier than tightening them).